src/System.Private.DataContractSerialization/src/System/Xml/XmlStreamNodeWriter.cs

```diff
@@ -419,12 +494,27 @@ protected virtual void FlushBuffer()
             }
         }
 
+        protected virtual async Task FlushBufferAsync()
+        {
+            if (_offset != 0)
+            {
+                await _stream.WriteAsync(_buffer, 0, _offset).ConfigureAwait(false);
+                _offset = 0;
+            }
```

Comment:
You can save some allocations by doing the following:
```csharp
if (_offset != 0)  
{
    var task = _stream.WriteAsync(_buffer, 0, _offset);
    _offset = 0;
    return task;
}
return Task.CompletedTask;
```
You could also think about removing the check for _offset!=0 and presume the caller has done this to avoid a method call/Task allocation unnecessarily. Replace it with a Debug.Assert to catch any code paths which don't do this.